### PR TITLE
Incorporate Spotbugs support into the agent

### PR DESCRIPTION
**Pull request by jtduffy**
Adds the ability to run Spotbugs static analysis on the agent via gradle or a GH action. The README contains the details on how to run the gradle command. The GHA is in a workflow named `GHA-Spotbugs.yml`:  

```
# Both HTML and XML files are generated for each project and attached to the workflow run page under 
# the "Artifacts" section. The files are downloaded as a zip (per project).  The XML file can be analyzed 
# inside the Spotbugs GUI: https://spotbugs.readthedocs.io/en/stable/running.html
```

The GHA currently only runs on demand. If the GHA is added to the standard workflow runs on a PR, the GHA will not fail the build under any circumstances (same for the gradle task).

The end goal being that we can run Spotbugs and determine as a team what findings should be added to the backlog for correction (possibly after each release to see if any new bugs pop up).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -49,6 +51,11 @@ subprojects {
         jcenter()
     }
 
+    def isSpotbugsEligible = ['newrelic-agent', 'newrelic-weaver', 'agent-bridge', 'agent-bridge-datastore', 'agent-model', 'agent-interfaces', 'newrelic-api'].contains(project.name)
+    if (isSpotbugsEligible) {
+        apply plugin: 'com.github.spotbugs'
+    }
+
```
Comment (meiao) [inline build.gradle:58]: Could the apply plugin be done in `spotbugs-config.gradle`?
This way only the modules that include that file will have it and we skip this check.

File: .github/workflows/GHA-Spotbugs.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      - name: Execute Spotbugs
+        env:
+          JAVA_HOME: ${{ env.ORG_GRADLE_PROJECT_jdk8 }}
+        run: |
+          ./gradlew $GRADLE_OPTIONS spotbugsMain --parallel
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: newrelic-agent-spotbugs-results
+          path: 'newrelic-agent/build/spotbugs/main.*'
```
[the remark this answers is not in the document]
Comment (jtduffy) [reply .github/workflows/GHA-Spotbugs.yml:69]: I tried the full capture glob initially, and it seems to present better with all of them split out. I'm not attached either way, so whatever everyone thinks.

File: .github/workflows/GHA-Spotbugs.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        required: false
+        default: 'main'
+        type: string
+
+jobs:
+  spotbugs:
+    timeout-minutes: 15
+    runs-on: ubuntu-20.04
+    permissions:
+      security-events: write
```
[the remark this answers is not in the document]
Comment (jtduffy) [reply .github/workflows/GHA-Spotbugs.yml:20]: Nope, left over from an attempt with another action

File: spotbugs-config.gradle [hunk 4]
```diff
@@ -0,0 +1,20 @@
+spotbugs {
```
[the remark this answers is not in the document]
Comment (jtduffy) [reply spotbugs-config.gradle]: done

File: build.gradle (hunk 1 above)
Comment (jtduffy) [reply build.gradle]: done

**Review by tbradellis — APPROVED** [review]

**Pull request merged**